orbstation/code/antagonists/contract_killer/contract_killer_gear.dm

```diff
@@ -0,0 +1,425 @@
+/// Sets of gear summonable by the Contract Killer to aid in their kill.
+
+/// Action that allows summoning a gear kit
+/datum/action/get_killer_gear
+	name = "Summon Equipment"
+	desc = "Offers you a choice of three gear kits in order to carry out your assassination. The kit will show up in your hand, \
+		so make sure you summon it when no one is looking."
+	check_flags = AB_CHECK_CONSCIOUS | AB_CHECK_INCAPACITATED | AB_CHECK_HANDS_BLOCKED
+	button_icon = 'icons/obj/storage/case.dmi'
+	button_icon_state = "secure"
+
+	var/list/equip_options = list() //list of kits to offer, set when the action is granted
+
+/// Set up which three kits the contract killer will be offered
+/datum/action/get_killer_gear/New()
+	..()
+	var/list/potential_kits = list()
+	for(var/datum/contract_killer_kit/kit as anything in subtypesof(/datum/contract_killer_kit))
+		potential_kits[kit] = initial(kit.weight)
+
+	for(var/i in 1 to 3)
+		var/new_kit = pick_weight(potential_kits)
+		potential_kits -= new_kit
+		equip_options += new_kit
+
+/datum/action/get_killer_gear/Trigger(trigger_flags)
+	. = ..()
+
+	var/list/kit_types = list()
+	for(var/datum/contract_killer_kit/kit as anything in equip_options)
+		kit_types[initial(kit.name)] = new kit
+
+	var/list/items = list()
+	for(var/kit_name in kit_types)
+		var/datum/contract_killer_kit/current_kit = kit_types[kit_name]
+		var/datum/radial_menu_choice/option = new
+		option.image = image(icon = current_kit.icon, icon_state = current_kit.icon_state)
+		option.info = span_boldnotice(current_kit.description)
+		items[kit_name] = option
```

Comment:
You might as well pregenerate this in `New` as you have everything you need for it